idunn/geocoder/models/params.py

```diff
@@ -19,8 +18,7 @@
 FOCUS_DECAY = float(settings["FOCUS_DECAY"])
 
 
-@dataclass
-class QueryParams:
+class QueryParams(BaseModel):
```

Comment:
Unfortunately, this causes unexpected changes about how FastAPI handles the parameters. `Query()` used as a default value in a pydantic model is ignored by FastAPI. As a result, all non-str types are considered as fields that should appear in the request body. And moreover, most of the documentation would be lost.

The good news is that it should be possible to keep most of your changes AND the `@dataclass`.
The `shape` field could indeed be defined with `Body(None,..., embed=True)`
See https://fastapi.tiangolo.com/tutorial/body-multiple-params/#embed-a-single-body-parameter

Reply:
Oh right, sorry about that, this is quite embarassing.

My main issue was that with this use of pydantic's dataclasses it is not possible to use the default constructor since the default values are in the form `Query(...)`. However it is possible to use `__pydantic_model__` to access inner model, which handles `Query` correctly :tada: 

This significant change of behavior by using a similar type which actually behaves strangely really seems very counter intuitive to use :disappointed: I suppose this behavior for dataclasses is slightly hacky and unintended ? (A quick search through the docs only raises the changelog from the date it was supported, which feels like it should only be an alternative to the use of models)